### ISSUE TITLE
`opam install --locked` silently ignores the --locked option

### ISSUE THREAD

**Issue by brendanlong** [ocaml/opam#4313]
The behavior of `install --locked` is really confusing, especially because there's no indication that anything is wrong.

The thing I would expect to work:

```bash
# Pin package(s) in the current repo but don't install deps yet
opam pin add -y -n .
# Install depexts
opam depext mypackage -y
# Now install deps
opam install . -y --locked
```

In this case, the `--locked` argument does absolutely nothing and the deps are installed only using mypackage.opam.

After reading `opam install --help`, I realized that `--locked` only works for with `opam pin`:

```bash
# Pin package(s) in the current repo but don't install deps yet
opam pin add -y -n .
# Install depexts
opam depext mypackage -y
# Now install deps in a really convoluted way
opam pin remove -y mypackage
opam pin add -y --locked .
```

There are two things I'd propose changing:

1. Please make `opam install --locked` use the lock file
2. Until (1) is done, make `opam install --locked` return an error message instead of silently ignoring the argument

**Comment by rjbou** [comment]
Opam install uses the lock file when the option is given. 
What opam version do you use?
Can you share the output of the install locked command? Also, you pin a local directory, is it a git or path pin?

**Comment by rjbou** [comment]
Any news on this one ?

**Comment by AltGr** [comment]
There seems to indeed be a bug here where the auto update done on `opam install DIR` doesn't handle the changes between the `.opam` file (which the package is currently pinned as) and the `.opam.locked` file:
```
% opam install . --locked
[NOTE] Ignoring uncommitted changes in /foo/bar (`--working-dir' not active).
[bar.0.10] synchronised (no changes)
```

**Comment by jvillard** [comment]
Not sure if it's the same issue (should I open a new one?) but I have another case where `--locked` doesn't use the lock file:

```
opam install --deps-only /path/to/opam/foo.opam --locked
```

This ignores the lock file in "/path/to/opam/foo.opam.locked". A workaround is to use `opam install --deps-only /path/to/opam/foo.opam.locked` instead but I'm quite sure the `--locked` command used to work when our opam files weren't in a separate opam/ directory so my guess is that opam doesn't look in the correct place for a .locked file?

**Comment by rjbou** [comment]
@jvillard As you give opam an opam file, it takes that opam file, and don't look for the locked one. With `opam install --deps-only /path/to/opam --locked`, it should be ok.

**Comment by rjbou** [comment]
The update/upgrade of locked pins is fixed by #5080, it'll close the main issue

**Comment by rjbou** [comment]
And the second, @jvillard one, is fixed by #5148
